ext/dom/parentnode.c

```diff
@@ -175,10 +175,7 @@ xmlNode* dom_zvals_to_fragment(php_libxml_ref_obj *document, xmlNode *contextNod
 				xmlSetTreeDoc(newNode, documentNode);
 
 				if (newNode->type == XML_ATTRIBUTE_NODE) {
-					xmlFree(fragment);
-
-					php_dom_throw_error(HIERARCHY_REQUEST_ERR, stricterror);
-					return NULL;
+					goto hierarchy_request_err;
```

Comment:
So there is another such case before opening the fragment if there is a missing document.
Don't think it makes sense to goto the new error handling block but it seems to always throw an exception (which is probably sensible).

So something to maybe do in master is to change the ``int`` type to ``bool`` as this is what it seems to expect (might be wrong tho)

Reply:
> So there is another such case before opening the fragment if there is a missing document.
Don't think it makes sense to goto the new error handling block but it seems to always throw an exception (which is probably sensible).

We can do kinda do it if we change the error block to first throw, then free, then return. Then put a second label just before the freeing. Then change the above check to a throw + goto the second label. Although that's a very minor cleanup of  course.

Like so:
```c
hierarchy_request_err:
	php_dom_throw_error(HIERARCHY_REQUEST_ERR, stricterror);
err: // <<< this is the second label
	xmlFreeNode(fragment);
	return NULL;
```

I think this would be a nice change because it's good to unify cleanup code to prevent bugs.

> So something to maybe do in master is to change the int type to bool as this is what it seems to expect (might be wrong tho)

You mean for stricterror? If so, yes sure I can do that during the merge.

Reply:
> > So there is another such case before opening the fragment if there is a missing document.
> > Don't think it makes sense to goto the new error handling block but it seems to always throw an exception (which is probably sensible).
> 
> We can do kinda do it if we change the error block to first throw, then free, then return. Then put a second label just before the freeing. Then change the above check to a throw + goto the second label. Although that's a very minor cleanup of course.
> 
> Like so:
> 
> ```c
> hierarchy_request_err:
> 	php_dom_throw_error(HIERARCHY_REQUEST_ERR, stricterror);
> err: // <<< this is the second label
> 	xmlFreeNode(fragment);
> 	return NULL;
> ```
> 
> I think this would be a nice change because it's good to unify cleanup code to prevent bugs.
> 

Then go for it :)

> > So something to maybe do in master is to change the int type to bool as this is what it seems to expect (might be wrong tho)
> 
> You mean for stricterror? If so, yes sure I can do that during the merge.

Yeah, although having this as a separate commit makes more sense as those are kinda unrelated changes